### ISSUE TITLE
Manual clamp variant for min and max

### ISSUE THREAD

**Issue by ronnodas** [rust-lang/rust-clippy#16884]
### What it does

Replaces manual implementations of `Ord::min` and `Ord::max` with those methods.

### Advantage

The code becomes more concise, (subjectively) clearer and doesn’t use any control flow.

### Drawbacks

In `const` context, primitives can be compared but the `Ord` methods cannot be called (as yet). The lint should not trigger in this case.

### Example

```rust
if x > y {
    y
} else {
    x
}
```

Could be written as:

```rust
x.min(y)
```

Note that if the comparison were `x >= y` then the suggested code should be `y.min(x)`.

### Comparison with existing lints

The two cases, min and max, can each be considered half of `manual_clamp`, and should probably just be part of that lint.

**Comment by Souradip121** [comment]
@rustbot claim

Implemented this as a new manual_min_max lint in the complexity category. The lint detects all eight if-else variants of manual Ord::min and Ord::max (covering both strict and non-strict comparisons, and both orderings of the then/else branches) and suggests the corresponding method call.

The implementation uses eq_expr_value to match the branches against the comparison operands, which also implicitly rejects expressions with side effects since it uses SpanlessEq::deny_side_effects internally. Floats are excluded by checking that the type implements Ord rather than just PartialOrd, which avoids the NaN semantic difference. The lint skips const contexts because Ord::min and Ord::max are not yet const-stable. Suggestions use Sugg::hir(...).maybe_paren() on the receiver so compound expressions like a + 1 are correctly parenthesized to (a + 1).min(b).

The applicability is MachineApplicable since, unlike manual_clamp, these methods never panic.
